### PR TITLE
Rename `Hydrate` -> `HydrationBoundary`

### LINKED ISSUE #4692: Rename `Hydrate` to `HydrationBoundary`
and remove `useHydrate` hook

### PR DESCRIPTION
**Pull request by prateek3255**
- Rename `Hydrate` -> `HydartionBoundary`
- Remove `useHydrate` hook
- Update migration guide

Fixes #4692

### CODE REVIEW

File: docs/react/guides/migrating-to-v5.md [hunk 1]
```diff
@@ -230,3 +230,22 @@ import { batch } from 'solid-js'
 
 notifyManager.setBatchNotifyFunction(batch)
 ```
+
+### `Hydrate` has been renamed to `HydrationBoundary` the `useHydrate` hook has been removed
```
Comment (TkDodo) [inline docs/react/guides/migrating-to-v5.md:234]:
```suggestion
### `Hydrate` has been renamed to `HydrationBoundary` and the `useHydrate` hook has been removed
```

**Review by TkDodo — APPROVED** [review]

**Comment by prateek3255** [comment]
@TkDodo Will open a different PR for codemod. Also wanted to ask if we need to update the existing documentation as well or just the migration guide?

**Comment by TkDodo** [comment]
yes, the docs as well please ❤️

**Comment by prateek3255** [comment]
Done @TkDodo

**Review by TkDodo — APPROVED** [review]

**Pull request merged**